impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -103,6 +103,12 @@ std::shared_ptr<ContextVK> ContextVK::Create(Settings settings) {
   return context;
 }
 
+size_t ChooseThreadCountForWorkers(size_t hardware_concurrency) {
+  // Never create more than 4 worker threads. Attempt to use up to
+  // half of the available concurrency.
+  return std::min(4ull, std::max(hardware_concurrency / 2ull, 1ull));
```

Comment:
nit: This would be much easier to parse with a `clamp` function

```
return clamp(hardware_concurrency / 2ull, 4ull, 1ull);
```

Reply:
![image](https://github.com/flutter/engine/assets/8975114/3bb8fabf-a6a5-46b2-9514-9373a6c25d60)


Give 'em the clamps.

Reply:
Done

Reply:
oh cool, I didn't know they have one in `std`.  I think the linter is going to complain that the arg names should be `lo` and `hi`... yea, `lo` and `hi` 🙄

Reply:
they charge per letter